File: CSharpAPI/Dto/Buy.cs
using System;
using System.Collections.Generic;

namespace Dto;

public partial class Buy
{
    public short Id { get; set; }

    public string? CodeClient { get; set; }

    public DateTime? Date { get; set; }

    public int? SumPrice { get; set; }

    public int? SumCount { get; set; }

    public string? Note { get; set; }


    public string? ClientName { get; set; }

    public List<Product>? PurchaseDetails { get; set; } = new List<Product>();

}

File: AngularApp/src/app/class/Buy.ts
import { Product } from "./product";
import { PurchaseDetails } from "./PurchaseDetail";

export class Buy{
    constructor(public id:number=0,
    public codeClient ?:string , 
    public date?:Date 
    ,public sumPrice:number=0 ,
    public sumCount:number=0
    ,public note ?:string 
     , public clientName?:string, public purchaseDetails:Array<Product>=new Array<Product>()){}
}